aspnetcore/signalr/authn-and-authz.md

```diff
@@ -0,0 +1,114 @@
+
+In the .NET client, there is a simlar `AccessTokenProvider` property that can be used to configure this:
+
+```csharp
+var connection = new HubConnectionBuilder()
+    .WithUrl("https://example.com/myhub", options => options.AccessTokenProvider = () => _myAccessToken)
+    .Build();
+```
+
+> [!NOTE]
+> This function is called before **every** HTTP request made by SignalR. This means that if you need to renew the token in order to keep the connection active (because it may expire during the connection) you can do so from within this function and SignalR will automatically start using the new token.
+
+In standard Web APIs, bearer tokens are sent via an HTTP header. However, due to security limitations imposed by browsers, the browser client is not able to set these headers when using the WebSockets or Server-Sent Events transports. When using those transports, the token is transmitted as a query string parameter. In order to support this on the server, additional configuration is required:
+
+[!code-csharp[Configure Server to accept access token from Query String](authn-and-authz/sample/Startup.cs?range=33-34,42-80,90)]
+
+### Windows Authentication
+
+If you have configured [Windows Authentication](xref:security/authentication/windowsauth) in your application, SignalR can use that authentication to secure Hubs. However, in order to send messages to individual users, you need to add a custom User ID provider, because the Windows Authentication system does not provide the "Name Identifier" claim that SignalR uses to determine the user name. To do this, add a new class that implements `IUserIdProvider` and retrieve one of the claims from the user to use as the identifier. For example, to use the "Name" claim (which is the Windows username in the form `[Domain]\[Username]`), create the following class:
+
+```csharp
+public class NameUserIdProvider : IUserIdProvider
+{
+    public string GetUserId(HubConnectionContext connection)
+    {
+        return connection.User?.FindFirst(ClaimTypes.Name)?.Value;
+    }
+}
+```
+
+Instead of `ClaimTypes.Name`, you can use any value from the `User` (such as the Windows SID identifier, etc.).
+
+> [!NOTE]
+> The value you choose must be unique among all the users in your system. Otherwise, a message intended for one user could end up going to a different user.
+
+Then, register this component in your `ConfigureServices` method **after** the call to `.AddSignalR`
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    // ... other services ...
+
+    services.AddSignalR();
+    services.AddSingleton<IUserIdProvider, NameUserIdProvider>();
+}
+```
+
+## Authorize users to access Hubs and Hub methods
+
+By default, all methods in a Hub can be called by an unauthenticated user. In order to require authentication, apply the [`Authorize`](/dotnet/api/microsoft.aspnetcore.authorization.authorizeattribute?view=aspnetcore-2.1) attribute to the Hub:
```

Comment:
Maybe point out that you can't require auth on all hubs like you used to be able to in ASP.NET SignalR

Reply:
I don't think we need to call out missing things like this here. Plus, we are considering bringing it back anyway :)